apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl

```diff
@@ -311,22 +313,43 @@ do_drop_db_v1(DB) ->
 do_store_batch_v1(DB, Shard, #{?tag := ?BATCH, ?batch_messages := Messages}, Options) ->
     emqx_ds_storage_layer:store_batch({DB, Shard}, Messages, Options).
 
+%% Remove me in EMQX 5.6
+-dialyzer({nowarn_function, do_get_streams_v1/4}).
 -spec do_get_streams_v1(
     emqx_ds:db(), emqx_ds_replication_layer:shard_id(), emqx_ds:topic_filter(), emqx_ds:time()
+) ->
+    [{integer(), emqx_ds_storage_layer:stream_v1()}].
+do_get_streams_v1(_DB, _Shard, _TopicFilter, _StartTime) ->
+    error(obsolete_api).
```

Comment:
Is the idea that we don't care about rolling upgrade stability for the time being?

Reply:
Yes. I added a disclaimer that new releases can currently break durable sessions in the documentation draft.